### ISSUE TITLE
Mod not working on server

### ISSUE THREAD

**Issue by Inpaladin** [ricksouth/serilum-mc-mods#864]
## **Information**
Minecraft version: 1.18.1
Modloader: Fabric
Fabric loader version: 0.12.12
Environment: Multiplayer, server

Mod name: Infinite Trading
Mod version: 1.18.x-2.1

## **Description**
World loads fine, the mod just doesn't do anything in-game.  Villagers still run out of stock.

**Comment by James103** [comment]
Can you please make sure that the latest versions of the following mods are installed server-side? All of them need to be there in order for Infinite Trading to work.
* Infinite Trading
* Collective
* Fabric API
* Fabric Loader

**Comment by Inpaladin** [comment]
currently running the latest version of all four minus fabric api which is one version out of date(new release 6 hours ago).  I don't want to restart the server to update the version right now because people are currently playing on it but I'll get back to you when I do.

**Comment by Inpaladin** [comment]
Updated the fabric api version on the server, still having the issue it seems.

**Comment by James103** [comment]
Next time a villager locks out their trades, can you please post the output from running the following command while near the offending villager?
`/execute as @e[type=villager,sort=nearest,distance=..128] run data get entity @s`

**Comment by Inpaladin** [comment]
spawned in a villager in a single player world and got this output:
![image](https://user-images.githubusercontent.com/59532097/147369648-9ed3ee17-6e06-4cc1-8331-ebbd0f7e1177.png)
![image](https://user-images.githubusercontent.com/59532097/147369658-fbef0b5a-3931-404a-bc54-b6baf0bee9a7.png)

**Comment by Inpaladin** [comment]
also, just for some additional information, here's the modlist the server is using
![image](https://user-images.githubusercontent.com/59532097/147373379-480550f1-2500-465c-b6bf-b485aa8368fe.png)

**Comment by chimericdream** [comment]
I was coming to report the same issue. I am seeing it in both single player and on a server. Here's all the usual data, plus the brain contents of the villager I tested. Looking at the trades list, this is what I would expect to see if the mod wasn't installed at all, so I am wondering if there's just a missing hook somewhere?

Minecraft: 1.18.1
Fabric: 0.12.12
Collective: 3.14
Infinite Trading: 2.1

(NBT formatted to make it easier to read, otherwise unchanged)

Entity data before trading:
```
{
  Brain: {
    memories: {
      "minecraft:job_site": {value: {pos: [I; -84, -55, 339], dimension: "minecraft:overworld"}}
    }
  },
  HurtByTimestamp: 0,
  Attributes: [
    {
      Base: 48.0d,
      Modifiers: [{Amount: 0.009601203065828231d, Operation: 1, UUID: [I; 639594596, -558151454, -1326682792, 934253217], Name: "Random spawn bonus"}],
      Name: "minecraft:generic.follow_range"
    },
    {
      Base: 0.5d,
      Name: "minecraft:generic.movement_speed"
    }
  ],
  FoodLevel: 0b,
  Invulnerable: 0b,
  FallFlying: 0b,
  ForcedAge: 0,
  Gossips: [],
  PortalCooldown: 0,
  AbsorptionAmount: 0.0f,
  LastRestock: 0L,
  FallDistance: 0.0f,
  DeathTime: 0s,
  Xp: 0,
  LastGossipDecay: 604958L,
  HandDropChances: [0.085f, 0.085f],
  PersistenceRequired: 0b,
  UUID: [I; 252499458, 841895516, -1074548294, -738053703],
  Age: 0,
  BalmData: {},
  Motion: [0.0d, -0.0784000015258789d, 0.0d],
  Health: 20.0f,
  LeftHanded: 0b,
  Air: 300s,
  OnGround: 1b,
  Offers: {
    Recipes: [
      {
        maxUses: 16,
        buyB: {id: "minecraft:air", Count: 1b},
        buy: {id: "minecraft:potato", Count: 26b},
        sell: {id: "minecraft:emerald", Count: 1b},
        xp: 2,
        uses: 0,
        priceMultiplier: 0.05f,
        specialPrice: 0,
        demand: 0,
        rewardExp: 1b
      },
      {
        maxUses: 16,
        buyB: {id: "minecraft:air", Count: 1b},
        buy: {id: "minecraft:emerald", Count: 1b},
        sell: {id: "minecraft:bread", Count: 6b},
        xp: 1,
        uses: 0,
        priceMultiplier: 0.05f,
        specialPrice: 0,
        demand: 0,
        rewardExp: 1b
      }
    ]
  },
  Rotation: [-32.05075f, 0.0f],
  HandItems: [{}, {}],
  RestocksToday: 0, 
  ArmorDropChances: [0.085f, 0.085f, 0.085f, 0.085f],
  Pos: [-82.5d, -55.0d, 339.5d],
  Fire: -1s,
  ArmorItems: [{}, {}, {}, {}],
  CanPickUpLoot: 1b,
  VillagerData: {profession: "minecraft:farmer", level: 1, type: "minecraft:plains"},
  HurtTime: 0s,
  Inventory: []
}
```

Entity data after trading:
```
{
  Brain: {
    memories: {
      "minecraft:job_site": {value: {pos: [I; -84, -55, 339], dimension: "minecraft:overworld"}}
    }
  },
  HurtByTimestamp: 0,
  Attributes: [
    {
      Base: 48.0d,
      Modifiers: [{Amount: 0.009601203065828231d, Operation: 1, UUID: [I; 639594596, -558151454, -1326682792, 934253217], Name: "Random spawn bonus"}],
      Name: "minecraft:generic.follow_range"
    },
    {
      Base: 0.5d,
      Name: "minecraft:generic.movement_speed"
    }
  ],
  FoodLevel: 0b,
  Invulnerable: 0b,
  FallFlying: 0b,
  ForcedAge: 0,
  Gossips: [{Target: [I; -1564523714, -1663349880, -1559951404, 1066179246], Type: "trading", Value: 2}],
  PortalCooldown: 0,
  AbsorptionAmount: 0.0f,
  LastRestock: 0L,
  FallDistance: 0.0f,
  DeathTime: 0s,
  Xp: 32,
  LastGossipDecay: 604958L,
  HandDropChances: [0.085f, 0.085f],
  PersistenceRequired: 0b,
  UUID: [I; 252499458, 841895516, -1074548294, -738053703],
  Age: 0,
  BalmData: {},
  Motion: [0.0d, -0.0784000015258789d, 0.0d],
  Health: 20.0f,
  LeftHanded: 0b,
  Air: 300s,
  OnGround: 1b,
  Offers: {
    Recipes: [
      {
        maxUses: 16,
        buyB: {id: "minecraft:air", Count: 1b},
        buy: {id: "minecraft:potato", Count: 26b},
        sell: {id: "minecraft:emerald", Count: 1b},
        xp: 2,
        uses: 16,
        priceMultiplier: 0.05f,
        specialPrice: 0,
        demand: 0,
        rewardExp: 1b
      },
      {
        maxUses: 16,
        buyB: {id: "minecraft:air", Count: 1b},
        buy: {id: "minecraft:emerald", Count: 1b},
        sell: {id: "minecraft:bread", Count: 6b},
        xp: 1,
        uses: 0,
        priceMultiplier: 0.05f,
        specialPrice: 0,
        demand: 0,
        rewardExp: 1b
      },
      {
        maxUses: 12,
        buyB: {id: "minecraft:air", Count: 1b},
        buy: {id: "minecraft:pumpkin", Count: 6b},
        sell: {id: "minecraft:emerald", Count: 1b},
        xp: 10,
        uses: 0,
        priceMultiplier: 0.05f,
        specialPrice: 0,
        demand: 0,
        rewardExp: 1b
      },
      {
        maxUses: 12,
        buyB: {id: "minecraft:air", Count: 1b},
        buy: {id: "minecraft:emerald", Count: 1b},
        sell: {id: "minecraft:pumpkin_pie", Count: 4b},
        xp: 5,
        uses: 0,
        priceMultiplier: 0.05f,
        specialPrice: 0,
        demand: 0,
        rewardExp: 1b
      }
    ]
  },
  Rotation: [-32.05075f, 0.0f],
  HandItems: [{}, {}],
  RestocksToday: 0,
  ArmorDropChances: [0.085f, 0.085f, 0.085f, 0.085f],
  Pos: [-82.5d, -55.0d, 339.5d],
  Fire: -1s,
  ArmorItems: [{}, {}, {}, {}],
  CanPickUpLoot: 1b,
  VillagerData: {profession: "minecraft:farmer", level: 2, type: "minecraft:plains"},
  HurtTime: 0s,
  ActiveEffects: [{Ambient: 0b, ShowIcon: 1b, ShowParticles: 1b, Duration: 160, Id: 10b, Amplifier: 0b}],
  Inventory: []
}
```

**Comment by Inpaladin** [comment]
glad it's not just me!

**Comment by Thiana** [comment]
This is some sort of weird mod conflict btw. I disabled everything but the Collective and Infinite trades mods and after restarting my world infinite trading worked again. The really weird part though... after I re-enabled my other mods infinite trading still works.

**Comment by Inpaladin** [comment]
could you send the modlist?

**Comment by Thiana** [comment]
<details>
<summary>Modlist</summary>

<p>
<pre>
AdvancedCompass-fabric-1.18-1.2.6.jar
alwaysawitherskull-fabric_1.18.x-1.4.jar
architectury-3.3.7.jar
automaticdoors-fabric_1.18.x-1.9.jar
balm-fabric-2.4.0+0.jar
betterthanmending-1.3.0.jar
bottledair-fabric_1.18.x-1.0.jar
bottleyourxp-fabric_1.18.x-1.6.jar
BowInfinityFix-1.18.x-fabric-rv11.jar
clienttweaks-fabric-1.18-7.0.0(1).jar
cloth-config-6.1.48-fabric.jar
collective-fabric-1.18.x-3.14.jar
Companion-1.18.1-fabric-1.0.0.jar
craftingtweaks-fabric-1.18-13.0.0(1).jar
cryingghasts-fabric_1.18.x-1.4.jar
cryingportals-fabric_1.18.x-1.2.jar
deathbackup-fabric_1.18.x-1.5.jar
deepslateinstamine_fabric_mc1.18.1-1.3.jar
despawningeggshatch-fabric_1.18.x-2.3.jar
diggusmaximus-1.5.2-1.18.jar
dragondropselytra-fabric_1.18.x-1.4.jar
DungeonsArise-1.18.1-2.1.49c-fabric-beta.jar
easyelytratakeoff-fabric_1.18.x-2.2.jar
EntityCulling-Fabric-1.3.3.jar
erodingstoneentities-fabric_1.18.x-2.5.jar
extractpoison-fabric_1.18.x-1.6.jar
fabric-api-0.45.1+1.18.jar
feature_nbt_deadlock_be_gone_fabric-1.0.0+1.18.1.jar
firespreadtweaks-fabric_1.18.x-1.2.jar
gravelminer-fabric-1.18-12.0.0(1).jar
grindstonesharpertools-fabric_1.18.x-1.6.jar
guifollowers-fabric_1.18.x-1.7.jar
healingcampfire-fabric_1.18.x-3.3.jar
hoetweaks-fabric_1.18.x-1.2.jar
huskspawn-fabric_1.18.x-1.4.jar
infinitetrading-fabric_1.18.x-2.1.jar
InventoryProfilesNext-fabric-1.18-1.2.1.jar
inventorytotem-fabric_1.18.x-1.5.jar
justenoughkeys-0.5.2-fabric.jar
justmobheads-fabric_1.18.x-5.2.jar
justplayerheads-fabric_1.18.x-1.8.jar
keepmysoiltilled-fabric_1.18.x-1.0.jar
Kiwi-1.18.1-fabric-5.2.2.jar
litematica-fabric-1.18.0-0.9.0.jar
lithium-fabric-mc1.18.1-0.7.7.jar
malilib-fabric-1.18.0-0.10.0-dev.26.jar
mineralchance-fabric_1.18.x-1.6.jar
minihud-fabric-1.18.0-0.19.0-dev.20211204.230657.jar
modmenu-3.0.1.jar
mooshroomspawn-fabric_1.18.x-1.5.jar
morezombievillagers-fabric_1.18.x-1.5.jar
MouseTweaks-fabric-mc1.18-2.20.jar
nametagtweaks-fabric_1.18.x-1.7.jar
nohostilesaroundcampfire-fabric_1.18.x-3.8.jar
no_null_processors_fabric-2.0.1+1.18.1.jar
OverpoweredMending-1.18.1-2.3.1.jar
paperbooks-fabric_1.18.x-1.4.jar
petnames-fabric_1.18.x-1.6.jar
phosphor-fabric-mc1.18.x-0.8.1.jar
quickshulker-1.3.3-1.18.jar
randombonemealflowers-fabric_1.18.x-1.7.jar
repurposed_structures_fabric-3.2.0+1.18.1.jar
respawningshulkers-fabric_1.18.x-1.6.jar
right-click-harvest-1.1.0+1.18.1-fabric.jar
RoughlyEnoughItems-7.1.357.jar
scaffoldingdropsnearby-fabric_1.18.x-1.5.jar
silencemobs-fabric_1.18.x-2.0.jar
SkyVillages-1.0.1-1.18.1-fabric.jar
sleepsooner-fabric_1.18.x-2.6.jar
strayspawn-fabric_1.18.x-1.5.jar
surfacemushrooms-fabric_1.18.x-1.2.jar
tntbreaksbedrock-fabric_1.18.x-1.8.jar
transcendingtrident-fabric_1.18.x-2.2.jar
trashslot-fabric-1.18-11.0.1.jar
villagerdeathmessages-fabric_1.18.x-2.2.jar
villagernames-fabric_1.18.x-3.5.jar
wooltweaks-fabric_1.18.x-1.4.jar
wthit-fabric-4.4.0.jar
Xaeros_Minimap_21.22.6_Fabric_1.18.jar
XaerosWorldMap_1.18.8_Fabric_1.18.jar
</pre>
</p>
</details>

**Comment by ricksouth** [comment]
This has been fixed in version 3.0 of Infinite Trading. It has completely been rewritten. How the offers are reset has changed and it now accounts for demand correctly. Thanks for giving the mod a try and opening the issue!

https://www.curseforge.com/minecraft/mc-mods/infinite-trading-fabric/
